gateway/gateway_integration_test.go

```diff
@@ -83,6 +83,53 @@ func TestWebhook(t *testing.T) {
 	requireJobsCount(t, postgresContainer.DB, "gw", jobsdb.Unprocessed.State, 1)
 }
 
+func TestDocsEndpoint(t *testing.T) {
+	bcServer := backendconfigtest.NewBuilder().
+		WithWorkspaceConfig(
+			backendconfigtest.NewConfigBuilder().
+				WithSource(
+					backendconfigtest.NewSourceBuilder().
+						WithID("source-1").
+						WithWriteKey("writekey-1").
+						WithSourceCategory("webhook").
+						WithSourceType("SeGment").
+						Build()).
+				Build()).
+		Build()
+	defer bcServer.Close()
+
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err)
+
+	postgresContainer, err := postgres.Setup(pool, t)
+	require.NoError(t, err)
+	transformerContainer, err := destination.SetupTransformer(pool, t)
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	gwPort, err := kithelper.GetFreePort()
+	require.NoError(t, err)
+
+	wg, ctx := errgroup.WithContext(ctx)
+	wg.Go(func() error {
+		err := runGateway(ctx, gwPort, postgresContainer, bcServer.URL, transformerContainer.TransformURL, t.TempDir())
+		if err != nil {
+			t.Logf("rudder-server exited with error: %v", err)
+		}
+		return err
+	})
+
+	url := fmt.Sprintf("http://localhost:%d", gwPort)
+	health.WaitUntilReady(ctx, t, url+"/health", 60*time.Second, 10*time.Millisecond, t.Name())
+	resp, err := http.Get(url + "/docs")
+	require.NoError(t, err, "it should be able to get the docs")
+	require.Equal(t, http.StatusOK, resp.StatusCode)
+	func() { kithttputil.CloseResponse(resp) }()
+	require.Equal(t, resp.Header.Get("Content-Type"), "text/html; charset=utf-8")
```

Comment:
We could have a `require.Contains` on the response body as well. Just something to make sure it has some content that it makes sense.